notebooks/xgboost-model/cat_encoder.py

```diff
@@ -0,0 +1,31 @@
+
+from sklearn import preprocessing
+from sklearn.preprocessing import StandardScaler
+import numpy as np
+
+class CatEncoder:
+    def __init__(self, cat_columns, data, normalize: bool=True):
+        self.cat_indexes = [data.columns.get_loc(name) for name in cat_columns]
+        self.num_indexes = [idx for idx in range(len(data.columns)) if idx not in self.cat_indexes]
+        self.encoder = preprocessing.OneHotEncoder()
+        self.encoder.fit(data[cat_columns])
+        self.num_columns = list(data.columns[self.num_indexes])
+        self.cat_columns = cat_columns
+        cat_transformed_names = self.encoder.get_feature_names(input_features=self.cat_columns)
+        self._transformed_column_names =  self.num_columns + list(cat_transformed_names)
+        if normalize:
+            self.normalizer = StandardScaler()
+            self.normalizer.fit(data.iloc[:, self.num_indexes])
+        else:
+            self.normalizer = None
+
+    def __call__(self, x):
+        numeric = x[:, self.num_indexes]
+        if self.normalizer is not None:
+            numeric = self.normalizer.transform(numeric)
+        categorical = self.encoder.transform(x[:, self.cat_indexes]).toarray()
+        return np.concatenate((numeric, categorical), axis=1)
+
+    @property
```

Comment:
Should probably use the same cat encoder (and in general try to standardize - we're very actively considering moving this into a toolkit utils library to avoid us having many slightly different versions proliferating all over the place).  In particular this one doesn't have the `cat_indexes_of_feature` method

Reply:
Makes sense,I’ll add the same encoder here as well